src/tools/Tool.ts

```diff
@@ -1,18 +1,168 @@
 import Konva from 'konva'
-import { CanvasElement, VueKonvaStage } from '@/types/Canvas'
-import { CustomEvent } from '@/util/PointerEventMapper'
+import {
+  AdditionData,
+  CanvasElement,
+  MoveData,
+  RemovalData,
+  RequestCanvasEntity,
+  VueKonvaStage
+} from '@/types/Canvas'
+import { CustomEvent, CustomStageEvent } from '@/util/PointerEventMapper'
+import store from '@/main'
+import { SocketActions } from '@/store/modules/socket'
+import { StageGetters } from '@/store/modules/stage'
+import { Point } from 'konva/types/Util'
+import { CanvasEntityActions, CanvasEntityGetters, CanvasEntityState } from '@/store/modules/canvasEntity'
+import { LayerGetters } from '@/store/modules/layer'
+import { ToolGetters, ToolsAction } from '@/store/modules/tools'
+import { CanvasAction, CanvasGetters } from '@/store/modules/canvas'
 
 export type CanvasDownAction = (event: CustomEvent, stage: VueKonvaStage) => void;
 export type CanvasMoveAction = (event: CustomEvent, stage: VueKonvaStage) => void;
 export type CanvasUpAction = (event: CustomEvent, stage: VueKonvaStage) => void;
-export type MouseDownAction = (event: CustomEvent, canvasElement: CanvasElement, layer: Konva.Layer, socket: WebSocket) => void;
-export type MouseMoveAction = (event: CustomEvent, canvasElement: CanvasElement, layer: Konva.Layer, socket: WebSocket) => void;
-export type MouseUpAction = (event: CustomEvent, canvasElement: CanvasElement, layer: Konva.Layer, socket: WebSocket) => void;
-export type RenderCanvas = (canvasElement: CanvasElement, layer: Konva.Layer, event: CustomEvent) => void;
-export type SendToWebSocket = (canvasElement: CanvasElement, socket: WebSocket) => void;
+export type MouseDownAction = (event: CustomEvent) => void;
+export type MouseMoveAction = (event: CustomEvent) => void;
+export type MouseUpAction = (event: CustomEvent) => void;
+export type RenderCanvas = (request: RequestCanvasEntity) => void;
+
+export enum Tracker {
+  ADDITION = 'addition',
+  REMOVAL = 'removal',
+  REDO = 'redo',
+  UNDO = 'undo',
+  MOVE = 'move'
+}
+
+export class ToolClass {
```

Comment:
This class supports the tools with helper functions like adding to the state, sending to sockets etc.